contracts/protocol/pool/PoolApeStaking.sol

```diff
@@ -489,6 +492,105 @@ contract PoolApeStaking is
         }
     }
 
+    struct CompoundPairedApeRewardLocalVar {
+        address xTokenAddress;
+        IERC721 bakcContract;
+        address bakcNToken;
+        uint256 balanceBefore;
+        uint256 balanceAfter;
+        uint256[] amounts;
+        address[] originArray;
+        uint256 totalAmount;
+        address tmpPositionOwner;
+    }
+
+    /// @inheritdoc IPoolApeStaking
+    function claimPairedApeRewardAndCompound(
+        address nftAsset,
+        address[] calldata users,
+        ApeCoinStaking.PairNft[][] calldata _nftPairs
+    ) external nonReentrant {
+        require(users.length == _nftPairs.length, "invalid parameter");
+        DataTypes.PoolStorage storage ps = poolStorage();
+
+        CompoundPairedApeRewardLocalVar memory localVar;
+        localVar.xTokenAddress = ps._reserves[nftAsset].xTokenAddress;
+        localVar.bakcContract = INTokenApeStaking(localVar.xTokenAddress)
+            .getBAKC();
+        localVar.bakcNToken = ps
+            ._reserves[address(localVar.bakcContract)]
+            .xTokenAddress;
+        localVar.balanceBefore = APE_COIN.balanceOf(address(this));
+        localVar.amounts = new uint256[](_nftPairs.length);
+
+        for (uint256 i = 0; i < _nftPairs.length; i++) {
+            ApeCoinStaking.PairNft[] calldata nftPair = _nftPairs[i];
+            localVar.originArray = new address[](nftPair.length);
+            for (uint256 j = 0; j < nftPair.length; j++) {
+                localVar.tmpPositionOwner = INToken(localVar.xTokenAddress)
```

Comment:
why do we need to store this in `localVar`?

Reply:
stack too deep